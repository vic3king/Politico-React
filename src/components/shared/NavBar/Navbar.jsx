import React from 'react';
import { Link } from 'react-router-dom';
import '../../../style/navbar.scss';
import PropTypes from 'prop-types';

const NavBar = ({ LiTagOne, LiTagTwo }) => {
  return (
    <React.Fragment>
      <header className="main-header">
        <div>
          <i className="fa fa-balance-scale fa-lg" />
          <Link to="/" className="main-header__brand">
            Politico
          </Link>
        </div>
        <nav className="main-nav">
          <ul className="main-nav__items">
            {LiTagOne}
            {LiTagTwo}
          </ul>
        </nav>
      </header>
    </React.Fragment>
  );
};
NavBar.defaultProps = {
  LiTagOne: (PropTypes.defaultProps = ''),
  LiTagTwo: (PropTypes.defaultProps = ''),
};
NavBar.propTypes = {
  LiTagOne: PropTypes.object,
  LiTagTwo: PropTypes.object,
};

export default NavBar;
