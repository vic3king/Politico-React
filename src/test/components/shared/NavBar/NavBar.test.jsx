import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../../../../components/shared/NavBar/Navbar';
import LiTag from '../../../../components/shared/Buttons/LI-tag';

describe('NavBar component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list tag', () => {
    const wrapper = shallow(
      <NavBar
        LiTagOne={<LiTag to="/signup" value="Sign Up" />}
        LiTagTwo={<LiTag to="/login" value="Login" />}
      />
    );
    expect(wrapper.find('header'));
    expect(wrapper.find('nav'));
    expect(wrapper.find('ul'));
    expect(wrapper.find('div'));
    expect(wrapper.hasClass('main-header'));
  });
});
