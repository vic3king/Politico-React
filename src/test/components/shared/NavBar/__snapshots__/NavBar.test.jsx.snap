// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    LiTagOne=""
    LiTagTwo=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <header
        className="main-header"
      >
        <div>
          <i
            className="fa fa-balance-scale fa-lg"
          />
          <Link
            className="main-header__brand"
            to="/"
          >
            Politico
          </Link>
        </div>
        <nav
          className="main-nav"
        >
          <ul
            className="main-nav__items"
          >
            
            
          </ul>
        </nav>
      </header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <i
              className="fa fa-balance-scale fa-lg"
            />
            <Link
              className="main-header__brand"
              to="/"
            >
              Politico
            </Link>
          </div>,
          <nav
            className="main-nav"
          >
            <ul
              className="main-nav__items"
            >
              
              
            </ul>
          </nav>,
        ],
        "className": "main-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-balance-scale fa-lg"
              />,
              <Link
                className="main-header__brand"
                to="/"
              >
                Politico
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-balance-scale fa-lg",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Politico",
                "className": "main-header__brand",
                "to": "/",
              },
              "ref": null,
              "rendered": "Politico",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="main-nav__items"
            >
              
              
            </ul>,
            "className": "main-nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "",
                "",
              ],
              "className": "main-nav__items",
            },
            "ref": null,
            "rendered": Array [
              "",
              "",
            ],
            "type": "ul",
          },
          "type": "nav",
        },
      ],
      "type": "header",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <header
          className="main-header"
        >
          <div>
            <i
              className="fa fa-balance-scale fa-lg"
            />
            <Link
              className="main-header__brand"
              to="/"
            >
              Politico
            </Link>
          </div>
          <nav
            className="main-nav"
          >
            <ul
              className="main-nav__items"
            >
              
              
            </ul>
          </nav>
        </header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <i
                className="fa fa-balance-scale fa-lg"
              />
              <Link
                className="main-header__brand"
                to="/"
              >
                Politico
              </Link>
            </div>,
            <nav
              className="main-nav"
            >
              <ul
                className="main-nav__items"
              >
                
                
              </ul>
            </nav>,
          ],
          "className": "main-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-balance-scale fa-lg"
                />,
                <Link
                  className="main-header__brand"
                  to="/"
                >
                  Politico
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-balance-scale fa-lg",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Politico",
                  "className": "main-header__brand",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Politico",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="main-nav__items"
              >
                
                
              </ul>,
              "className": "main-nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "",
                  "",
                ],
                "className": "main-nav__items",
              },
              "ref": null,
              "rendered": Array [
                "",
                "",
              ],
              "type": "ul",
            },
            "type": "nav",
          },
        ],
        "type": "header",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
